polus-imagej-plugins/polus-imagej-image-integral-plugin/src/ij_converter.py

```diff
@@ -0,0 +1,118 @@
+
+'''
+A conversion utility built to convert abstract to primitive
+This works for Threshold Apply and Gaussian Filter functions
+Note change from jnius to jpype for handling conversions in ij_converter.py
+
+'''
+
+import imagej
+import logging
+import imglyb
+import jpype
+import numpy as np
+import scyjava
+
+# Initialize the logger
+logging.basicConfig(format='%(asctime)s - %(name)-8s - %(levelname)-8s - %(message)s',
+                    datefmt='%d-%b-%y %H:%M:%S')
+logger = logging.getLogger("ij_converter")
+logger.setLevel(logging.INFO)
+
+ij = None
+
+## fill in types to convert
+ABSTRACT_ITERABLES = [
+    'IterableInterval',
+    'Iterable'
+]
+
+ABSTRACT_SCALARS = [
+    'RealType'
+]
+
+SCALARS = [
+    'double'
+]
+
+## recognize array objects as scalar objects + '[]'
+ARRAYS = [s+'[]' for s in SCALARS]
+
+def _java_setup():
+    global IMGLYB_PRIMITIVES, PRIMITIVES, PRIMITIVE_ARRAYS
+    IMGLYB_PRIMITIVES = {
+        'float32'    : imglyb.types.FloatType,
+        'float64'    : imglyb.types.DoubleType,
+        'int8'       : imglyb.types.ByteType,
+        'int16'      : imglyb.types.ShortType,
+        'int32'      : imglyb.types.IntType,
+        'int64'      : imglyb.types.LongType,
+        'uint8'      : imglyb.types.UnsignedByteType,
+        'uint16'     : imglyb.types.UnsignedShortType,
+        'uint32'     : imglyb.types.UnsignedIntType,
+        'uint64'     : imglyb.types.UnsignedLongType
+    }
+    PRIMITIVES = {
+        'double'     : jpype.JDouble,
+        'float'      : jpype.JFloat,
+        'long'       : jpype.JLong,
+        'int'        : jpype.JInt,
+        'short'      : jpype.JShort,
+        'char'       : jpype.JChar,
+        'byte'       : jpype.JByte,
+        'boolean'    : jpype.JBoolean
+    }
+    PRIMITIVE_ARRAYS = {
+        'double[]'     : jpype.JDouble[:],
+        'float[]'      : jpype.JFloat[:],
+        'long[]'       : jpype.JLong[:],
+        'int[]'        : jpype.JInt[:],
+        'short[]'      : jpype.JShort[:],
+        'char[]'       : jpype.JChar[:],
+        'byte[]'       : jpype.JByte[:],
+        'boolean[]'    : jpype.JBoolean[:]
+    }
+scyjava.when_jvm_starts(_java_setup)
+
+JAVA_CONVERT = {}
+JAVA_CONVERT.update({
+    t:lambda s,t,st: IMGLYB_PRIMITIVES[str(st)](st.type(s)) for t in ABSTRACT_SCALARS
+})
+JAVA_CONVERT.update({
+    t:lambda s,t,st: PRIMITIVES[t](float(s)) for t in SCALARS
+})
+JAVA_CONVERT.update({
+    t:lambda s,t,st: PRIMITIVE_ARRAYS[t]([float(si) for si in s.split(',')]) for t in ARRAYS
+})
+JAVA_CONVERT.update({
+    t:lambda s: imglyb.util.Views.iterable(ij.py.to_java(s)) for t in ABSTRACT_ITERABLES
+})
+
+def to_java(np_array,java_type,java_dtype=None):
+
+    if ij == None:
+        raise ValueError('No imagej instance found.')
+
+    if isinstance(np_array,type(None)):
+        return None
+
+    if java_type in JAVA_CONVERT.keys():
+        if java_dtype != None:
```

Comment:
What's going on here is tricky. The function `to_java` taking responsibility for the arguments of the `JAVA_CONVERT` lambda functions. But there is an implicit bit of logic going on that is problematic. 

1. This code was written assuming the caller of `to_java` knows which types require three arguments, and which types don't. i.e. the caller should know that scalars and arrays but not interables require `java_dtype`. 
2. This code also needs to have knowledge of how the lambdas were constructed. Which it shouldn't. 

Instead, I recommend doing the following:
1. For lambdas requiring three inputs, that should be reduced to two. `java_type` is both a key to `JAVA_CONVERT` and also an argument to the lambda. That eans in the definition above it can be hard coded in the list comprehension and therefore only require two input arguments: `np_array` and `java_dtype`. 
2. The function `to_java` should take in `np_array`, `java_type`, and `*args` which it passes on to the lambda. This way all lambdas can be called via a unified signature.

This function also requires documentation since it's not clear how it works and what the expected inputs are. A couple basic examples in the docstring of utilizing the different type of conversion would be very useful. Otherwise, I'm left wondering why this is needed at all if the call on line 106 works just fine.